picard/script/serializer.py

```diff
@@ -166,6 +166,38 @@ def update_from_dict(self, settings):
         if updated and 'last_updated' not in settings:
             self.update_last_updated()
 
+    def to_dict(self):
+        """Generate a dictionary containing the object's OUTPUT_FIELDS.
+
+        Returns:
+            dict: Dictionary of the object's OUTPUT_FIELDS
+        """
+        items = {key: getattr(self, key) for key in self.OUTPUT_FIELDS}
+        items["script"] = str(items["script"])
+        return items
+
+    @classmethod
+    def create_from_dict(cls, script_dict, create_new_id=True):
+        """Creates an instance based on the contents of the dictionary provided.
+        Properties in the dictionary that are not found in the script object are ignored.
+
+        Args:
+            script_dict (dict): Dictionary containing the property settings.
+            create_new_id (bool, optional): Determines whether a new ID is generated. Defaults to True.
+
+        Returns:
+            object: An instance of the class, populated from the property settings in the dictionary provided.
+        """
+        new_object = cls()
+        if not isinstance(script_dict, dict):
```

Comment:
`isinstance(script_dict, Mapping)` is more pythonic (any object acting as dict can be used).
https://www.python.org/dev/peps/pep-3119/#mappings

Reply:
Now I remember you mentioning that to me before.  I had forgotten.  Sorry about that.  I'll make the change.